### PR TITLE
Add proposal template

### PR DESCRIPTION
**Pull request by emilypi**
This MR adds a tech proposal template, and beefier README. It also includes a license.

### CODE REVIEW

File: TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+This template is for Haskell Foundation technical agenda proposals. You may find existing proposals and more examples at the [tech-proposals](https://github.com/haskellfoundation/tech-proposals) repository. 
+
+
+
+## Introduction
+
+This section should detail the following: A summary of the proposal’s
+motivation, implementation, risks, timeline, and goals. It should aim to
+be an abstract for the proposals.
```
Comment (goldfirere) [inline TEMPLATE.md:13]:
```suggestion
be an abstract for the proposal.
```

File: TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+This section should detail the high-level goals of the project, as well
+as the criteria for what constitutes a finished project. Deliverables
+should support these goals, and the goals should support the motivation
+for undertaking the project.
+
+## People
+
+This section should detail the following people:
+
+-   **Performers:** the project leads and main developers/resources who will be directly involved in the day to day work.
```
Comment (goldfirere) [inline TEMPLATE.md:40]:
```suggestion
-   **Performers:** the project leads and main developers/resources who will be directly involved in the day to day work. If your project needs performers to be sourced by the Haskell Foundation, please say so here.
```

I would want the possibility of our finding volunteers for a worthwhile project to be explicitly possible. Of course, a proposal with performers attached is more compelling than one without.

File: TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+## Timeline
+
+This section should detail the high-level time estimate of such a
+project, and serve as an educated guess. We acknowledge that no estimate
+is accurate, but we should be able to give coarse estimates in terms of
+months and years.
+
+## Lifecycle:
+
+This section should detail the SDLC of the project, including a
```
Comment (goldfirere) [inline TEMPLATE.md:61]: SDLC? My guess is "software development lifecycle", but this is an acronym I'm unfamiliar with.

File: TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    stakeholders or performers.
+    
+-   **Performance impact:** An overview of the potential performance impact of this project on the ecosystem, if applicable.
+
+## Deliverables
+
+This section should describe the individual deliverables this project
+will bring as a result of the changes described above. These
+deliverables should support the end goals of the project and define what
+HF is actually delivering to the community.
```
Comment (goldfirere) [inline TEMPLATE.md:77]:
```suggestion
this project is actually delivering to the community.
```

The HF might deliver organization, for example, while the project delivers a new wizwoz to be uploaded to Hackage.

**Review by goldfirere — COMMENTED** [review]
Looks great! Just a few small suggestions.

File: TEMPLATE.md (hunk 3 above)
Comment (emilypi) [reply TEMPLATE.md:61]: i'll expand. It's actually not correct in retrospect. It should just be "project lifecycle"

**Pull request merged**